engine/resources/cron.go

```diff
@@ -462,17 +460,37 @@ func (obj *CronUID) IFF(uid engine.ResUID) bool {
 	if !ok {
 		return false
 	}
-	return obj.name == res.name
+	if obj.unit != res.unit {
+		return false
+	}
+	if obj.session != res.session {
+		return false
+	}
+	return true
+}
+
+// AutoEdges returns the AutoEdge interface.
+func (obj *CronRes) AutoEdges() (engine.AutoEdge, error) {
+	return nil, nil
 }
 
 // UIDs includes all params to make a unique identification of this object.
 // Most resources only return one although some resources can return multiple.
 func (obj *CronRes) UIDs() []engine.ResUID {
-	x := &CronUID{
+	unit := fmt.Sprintf("%s.service", obj.Name())
+	if obj.Unit != "" {
+		unit = obj.Unit
+	}
+	cronUID := &CronUID{
 		BaseUID: engine.BaseUID{Name: obj.Name(), Kind: obj.Kind()},
-		name:    obj.Name(), // svc name
+		unit:    unit,        // name of target unit
+		session: obj.Session, // user session
+	}
+	file, err := obj.makeComposite()
```

Comment:
I think there's a more elegant way to write this...
uids := the one that's always there..
if f, err != nil {
uids=append(uids, f.uids())
return uids